packages/wifi_iot/android/src/main/java/com/alternadom/wifiiot/WifiIotPlugin.java

```diff
@@ -391,10 +401,19 @@ private void isSSIDHidden(Result poResult) {
 
       poResult.error("Exception [isSSIDHidden]", "Wifi AP not Supported", null);
     } else {
-      poResult.error(
-          "Exception [isSSIDHidden]",
-          "Getting SSID visibility is not supported on API level >= 26",
-          null);
+      if (apReservation != null) {
+        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.R) {
+          SoftApConfiguration softApConfiguration = apReservation.getSoftApConfiguration();
+          poResult.success(softApConfiguration.isHiddenSsid());
+        } else {
+          poResult.error(
+              "Exception [isSSIDHidden]",
+              "Getting SoftAPConfiguration is not supported on API level < 30",
+              null);
```

Comment:
Should we always return `true` for SDK 27 to 29? With some warning in logs? Instead of failing.

Reply:
Do you think the SSID is by default hidden ? If so yes adding a return `true` with a warning would be perfect.

Reply:
I found this - https://developer.android.com/reference/android/net/wifi/WifiConfiguration#hiddenSSID

Is that useful? `apReservation. getWifiApConfiguration().hiddenSSID` ??

Reply:
I think if it returns `null`, then we can assume `true`.

Reply:
My bad, I meant `false` - default LOHS as far as i know are not "hidden"

Reply:
> My bad, I meant `false` - default LOHS as far as i know are not "hidden"

@PoloLacoste You can test this and confirm it.

Reply:
> 

We are using this code for SDK API below 26. On higher SDK version we need a system level protected permission, see [here](https://stackoverflow.com/questions/46392277/changing-android-hotspot-settings).

Reply:
Oops, I meant [`LocalOnlyHotspotReservation.getWifiConfiguration()`]( https://developer.android.com/reference/android/net/wifi/WifiManager.LocalOnlyHotspotReservation#getWifiConfiguration()), the one we are already using for API 26-30:

`apReservation.getWifiConfiguration().hiddenSSID`

Reply:
Also have we considered `apReservation.getWifiConfiguration()` returning `null`? Check below quote from the link shared above.

> Returns the [`WifiConfiguration`](https://developer.android.com/reference/android/net/wifi/WifiConfiguration) of the current Local Only Hotspot (LOHS). May be null if hotspot enabled and security type is not `WifiConfiguration.KeyMgmt.None` or `WifiConfiguration.KeyMgmt.WPA2_PSK`.

Reply:
> Oops, I meant [`LocalOnlyHotspotReservation.getWifiConfiguration()`](https://developer.android.com/reference/android/net/wifi/WifiManager.LocalOnlyHotspotReservation#getWifiConfiguration()), the one we are already using for API 26-30:
> 
> `apReservation.getWifiConfiguration().hiddenSSID`

This should do the trick!

Reply:
> Also have we considered `apReservation.getWifiConfiguration()` returning `null`? Check below quote from the link shared above.
> 
> > Returns the [`WifiConfiguration`](https://developer.android.com/reference/android/net/wifi/WifiConfiguration) of the current Local Only Hotspot (LOHS). May be null if hotspot enabled and security type is not `WifiConfiguration.KeyMgmt.None` or `WifiConfiguration.KeyMgmt.WPA2_PSK`.

Yes we should return and error on a `null` object (`poResult.error`). Should be done!

Reply:
@PoloLacoste Handle it everywhere we use it, should we throw error or return `null` with warning? what do you think will be right?

Reply:
I think an error should be better, we are already throwing some errors if the Hotspot is not enabled.

Reply:
Ok